libsolidity/grammar.txt

```diff
@@ -5,44 +5,50 @@ ContractPart = StateVariableDeclaration | StructDefinition | ModifierDefinition
 
 InheritanceSpecifier = Identifier ( '(' Expression ( ',' Expression )* ')' )?
 StructDefinition = 'struct' Identifier '{'
-                     ( VariableDeclaration (';' VariableDeclaration)* )? '}
-StateVariableDeclaration = TypeName ( 'public' | 'inheritable' | 'private' )? Identifier ';'
+                     ( VariableDeclaration (';' VariableDeclaration)* )? '}'
+StateVariableDeclaration = TypeName ( 'public' | 'internal' | 'private' )? Identifier ';'
 ModifierDefinition = 'modifier' Identifier ParameterList? Block
 FunctionDefinition = 'function' Identifier ParameterList
-					 ( Identifier | 'constant' | 'external' | 'public' | 'inheritable' | 'private' )*
+                     ( Identifier | 'constant' | 'external' | 'public' | 'internal' | 'private' )*
                      ( 'returns' ParameterList )? Block
 
 EnumValue = Identifier
-EnumDefinition = 'enum' '{' EnumValue (',' EnumValue)* '}'
+EnumDefinition = 'enum' Identifier '{' EnumValue? (',' EnumValue)* '}'
 ParameterList = '(' ( VariableDeclaration (',' VariableDeclaration)* )? ')'
 // semantic restriction: mappings and structs (recursively) containing mappings
 // are not allowed in argument lists
 VariableDeclaration = TypeName Identifier
 TypeName = ElementaryTypeName | Identifier | Mapping | ArrayTypeName
 Mapping = 'mapping' '(' ElementaryTypeName '=>' TypeName ')'
-ArrayTypeName = TypeName '[' (Expression)? ']'
+ArrayTypeName = TypeName ('[' Expression? ']')+
 
 Block = '{' Statement* '}'
 Statement = IfStatement | WhileStatement | Block |
-			( Continue | Break | Return | VariableDefinition | ExpressionStatement ) ';'
+            ( Continue | Break | Return | Throw | VariableDefinition | ExpressionStatement ) ';'
 
 ExpressionStatement = Expression
 IfStatement = 'if' '(' Expression ')' Statement ( 'else' Statement )?
 WhileStatement = 'while' '(' Expression ')' Statement
 VardefOrExprStmt = Variabledefinition | ExpressionStatement
 ForStatement = 'for' '(' (VardefOrExprStmt)? ';' (Expression)? ';' (ExpressionStatement)? ')' Statement
-Continue = 'continue' ';'
-Break = 'break' ';'
-Return = 'return' Expression? ';'
-Throw = 'throw' Expression? ';'
-VariableDefinition = VariableDeclaration ( = Expression )? ';'
+Continue = 'continue'
+Break = 'break'
+Return = 'return' Expression?
+Throw = 'throw' Expression?
```

Comment:
Fixed. It was a leftover from the old `grammar.txt`. I have removed it in commit 447797ad9c5228ca50a4fa5b17ced4aba07e8d65